Duplicati/Library/Backend/S3/S3Backend.cs

```diff
@@ -218,7 +224,7 @@ public S3(string url, Dictionary<string, string> options)
 
                 //Change in S3, now requires that you use location specific endpoint
                 if (!string.IsNullOrEmpty(locationConstraint))
-                    foreach (KeyValuePair<string, string> kvp in DEFAULT_S3_LOCATION_BASED_HOSTS)
+                    foreach (var kvp in DEFAULT_S3_LOCATION_BASED_HOSTS)

```

Comment:
Minor point, but can we now perform a single lookup via `Dictionary.TryGetValue` instead of enumerating the entries?